pkg/algorithm/nodeselect/v1alpha1/select_node.go

```diff
@@ -263,6 +276,13 @@ func (ac *Config) getBlockDevice() (*ndmapis.BlockDeviceList, error) {
 	if len(bdl.Items) == 0 {
 		return nil, errors.Errorf("type {%s} devices are not available to provision pools in %s mode", diskType, ProvisioningType(ac.Spc))
 	}
+	if ProvisioningType(ac.Spc) == ProvisioningTypeAuto {
+		newBDL, err := bdl.GetUsableBlockDevices(ac.Spc.Name, ac.Namespace)
```

Comment:
why we need this? in what cases, this helps us

Reply:
When user created `n spc` at a time then controller will process one by one. Let us assume that `spc1` is processed and cstor-operator created claim when cstor-operator picked up second `spc2` to create pools then we shouldn't include the `bd` which is claimed for `spc-1`.

Reply:
if claims are not fulfilled, still, there are chances of picking a BD for 2 SPCs right?

Reply:
not fulfilled means not yet bounded? In this case Yes there are chances of getting that blockdevice but before using to create a pool we have strict check it should be claimed by spc which is processing.

Reply:
yes.. I meant - not fulfilled as 'not yet bound'
and I also saw other code where checks are there that BDs are claimed by SPC before creating pool
If CSP is already created, then, those BDs are already filtered out.
I'm trying to understand the need of this code.
This code will NOT cover the case that I mentioned above.
And, this code might cover the case where BDs are claimed but CSPs are not created yet. Is my understanding right?

Reply:
> If CSP is already created, then, those BDs are already filtered out.

No we are doing this filter in `getCandidateNode`

> And, this code might cover the case where BDs are claimed but CSPs are not created yet. Is my understanding right?

Yes this will cover above case.